src/app/components/jsx-helpers/jsx-helpers.jsx

```diff
@@ -159,12 +159,13 @@ export function RawHTML({Tag='div', html, embed=false, ...otherProps}) {
 
 export function useToggle(initialState) {
     const [value, setValue] = useState(initialState);
+    const toggleRef = useRef(
+        (newValue=!value) => {
+            setValue(newValue);
+        }
+    );
```

Comment:
it looks like you're using useRef to remember the function so it's not re-created on each render.  For that  you typically use either `useMemo` or `useCallback` https://reactjs.org/docs/hooks-reference.html#usememo

Reply:
It doesn't seem to work using useCallback because the function needs to take an argument. useRef docs explicitly say they are useful for any mutable value. I'll keep my eyes open for proper uses of useCallback and useMemo.